velox/connectors/hive/FileHandleKey.h

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cstdint>
+#include <functional>
+#include <string>
+
+namespace facebook::velox {
+class FileHandleKey {
+ public:
+  FileHandleKey(
+      int64_t fileSize_,
+      int64_t modificationTime_,
+      const std::string& filePath_)
+      : fileSize(fileSize_),
+        modificationTime(modificationTime_),
+        filePath(filePath_) {}
+
+  FileHandleKey(const std::string& filePath_) : filePath(filePath_) {}
+
+  std::string getFilePath() const {
+    return filePath;
+  }
+
+  int64_t getFileSize() const {
+    return fileSize;
+  }
+
+  int64_t getModificationTime() const {
+    return modificationTime;
+  }
+
+  bool operator==(const FileHandleKey& other) const {
+    return fileSize == other.fileSize &&
+        modificationTime == other.modificationTime &&
+        filePath == other.filePath;
+  }
+
+ private:
+  int64_t fileSize{-1};
+  int64_t modificationTime{-1};
+  std::string filePath{""};
+};
+} // namespace facebook::velox
+
+template <class T>
+inline void hash_combine(std::size_t& seed, const T& v) {
+  std::hash<T> hasher;
+  seed ^= hasher(v) + 0x9e3779b9 + (seed << 6) + (seed >> 2);
```

Comment:
Can you add a comment for the source of this?

Reply:
Done.